p2p-media-loader-hlsjs/lib/hlsjs-loader.ts

```diff
@@ -32,40 +69,56 @@ export class HlsJsLoader {
         _config: LoaderConfiguration,
         callbacks: LoaderCallbacks<LoaderContext>
     ): Promise<void> {
+        const start = performance.now();
+        this.stats.loading.start = start - DEFAULT_DOWNLOAD_LATENCY;
+        this.stats.loading.first = start;
+
         if (((context as unknown) as { type: unknown }).type) {
             try {
                 const result = await this.segmentManager.loadPlaylist(context.url);
+                this.isLoaded = true;
                 this.successPlaylist(result, context, callbacks);
             } catch (e) {
                 this.error(e, context, callbacks);
             }
         } else if (((context as unknown) as { frag: unknown }).frag) {
+            const { loader } = this.segmentManager;
+
+            const updateBandwidthEstimate = () => {
+                const bandwidthEstimate = loader.getBandwidthEstimate();
+                // convert bytes per millisecond to bits per second
+                this.stats.bwEstimate = bandwidthEstimate * 8000;
+                this.stats.loaded = (performance.now() - start) * bandwidthEstimate;
+            };
+            loader.on(Events.PieceBytesDownloaded, updateBandwidthEstimate);
```

Comment:
I think it should be a `setInterval` to ~250ms because stats may not be updated before the abandon rules check (and so stats.loaded === 0)